common/djangoapps/entitlements/tasks.py

```diff
@@ -62,3 +72,49 @@ def expire_old_entitlements(self, start, end, logid='...'):
         raise self.retry(exc=exc, countdown=countdown, max_retries=MAX_RETRIES)
 
     LOGGER.info('Successfully completed the task expire_old_entitlements after examining %d entries [%s]', entitlements.count(), logid)  # lint-amnesty, pylint: disable=line-too-long
+
+
+@shared_task(bind=True, ignore_result=True)
+@set_code_owner_attribute
+def expire_and_create_entitlements(self, no_of_entitlements):
```

Comment:
See comment below about race condition, but one option might be to mimic the arguments of `expire_old_entitlements`.

We would run the `CourseEntitlement.objects.filter` in the management command expire_and_create_entitlements.py and have each task operate on a range of entitlement ids.

Reply:
Done by @zubair-ce07 in 889ae7bc9c667cf61e6eed7d8af60c91e61ae381.